### PR TITLE
[Bugfix][MiniCPM-o] Guard native duplex silence scenario

### PR DESCRIPTION
**Pull request by ZacheryAU**
## Purpose
#6099 shows there was a crash for MiniCPM-o 4.5 in native duplex scenario when the model keeps silent and it was fixed in #6170. Specific unit tests are needed to guard that native duplex runs normally when the model should be silent.

## Summary
- Adds CPU unit tests for MiniCPM-o 4.5 native duplex `llm2tts` when Stage 0 emits a listen / control / terminal unit with neither `latent` nor `hidden_states`.
- Regression for #6099: a no-speech Thinker decision must skip the Talker handoff instead of raising `ValueError: No latent or hidden_states found in thinker output`.
- Does not change production code. The skip path landed in #6170; these tests lock it.

## Tests
`tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py`

1. **`test_native_duplex_control_only_without_hidden_states_skips_talker`**
   Parametrized non-empty control/terminal units: `[listen]`, `[chunk_eos]`, `[turn_eos]`, and `[listen, chunk_eos]`. These exercise the helper that classifies every token as control-only (the empty-output case does not). Each case must return `[]`.

2. **`test_native_duplex_empty_unit_without_hidden_states_skips_talker`**
   Empty native-duplex output (control/terminal with no token ids) and no hidden states. Same skip: do not wake the Talker.

3. **`test_native_duplex_listen_after_speak_without_hidden_states_skips_talker`**
   After a spoken segment, cumulative ids still contain the earlier speak tokens, but the streaming cursor has already handed those off. The current segment is listen-only and has no hidden states. `llm2tts` must skip that delta rather than require hidden states for the whole cumulative sequence.

4. **`test_native_duplex_speak_without_hidden_states_still_raises`**
   (Additional) Fail-closed contrast: a speak/content unit missing hidden states is a malformed speech handoff and must still raise. Skip is only for confirmed control-only / no-speech units.

5. **`test_ordinary_empty_output_without_hidden_states_still_raises`**
   (Additional) Skip is restricted to native duplex. An ordinary request with no duplex marker, empty token ids, and no hidden states must still raise. The existing adjacent missing-hidden test uses a content token, so it does not cover this boundary.

## Test plan

**vLLM Version:** 0.29.0

**vLLM-Omni Commit:** current commit on top of 538eddd74d57ac5a8fea0a8ac0d5288e20678ccb

- run `pytest -q tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py`. Tests 1-5 pass.
- copy the file onto `009b992c^` (pre-#6170). Tests 1–3 fail with the original `ValueError`; test 4-5 still pass.

### current commit
```
8 passed, 15 warnings in 0.09s
```
###  `009b992c^` (pre-#6170)
```bash
FAILED tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py::test_native_duplex_control_only_without_hidden_states_skips_talker[listen] - ValueError: No latent or hidden_states found in thinker output
FAILED tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py::test_native_duplex_control_only_without_hidden_states_skips_talker[chunk_eos] - ValueError: No latent or hidden_states found in thinker output
FAILED tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py::test_native_duplex_control_only_without_hidden_states_skips_talker[turn_eos] - ValueError: No latent or hidden_states found in thinker output
FAILED tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py::test_native_duplex_control_only_without_hidden_states_skips_talker[listen_then_chunk_eos] - ValueError: No latent or hidden_states found in thinker output
FAILED tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py::test_native_duplex_empty_unit_without_hidden_states_skips_talker - ValueError: No latent or hidden_states found in thinker output
FAILED tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py::test_native_duplex_listen_after_speak_without_hidden_states_skips_talker - ValueError: No latent or hidden_states found in thinker output
6 failed, 2 passed, 15 warnings in 0.29s
```

### CODE REVIEW

**Review by ZacheryAU — COMMENTED** [review]
self review (TL;DR):
- 1st test: should pass when only `listen` or `eos`
- 2nd test: should pass when empty
- 3rd test: should pass when `listen` even just spoke
- 4th test: should raise error when `speak` but no latent / hidden_states
- 5th test: should raise error when non-duplex but no latent / hidden_states

File: tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+
+def test_native_duplex_empty_unit_without_hidden_states_skips_talker() -> None:
+    """An empty native-duplex output with no speech payload must also skip.
+
+    Control/terminal units can arrive with no generated ids and no hidden
+    states. Same skip contract as listen-only: do not wake the Talker.
+    """
+    assert llm2tts([_native_duplex_thinker_output([])], prompt=None) == []
+
```
Comment (amy-why-3459) [inline tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py:117]: Non-blocking: could we also parameterize actual non-empty terminal/control units?

The empty-output case exercises the helper's immediate `if not output_ids: return True` branch, so it does not verify that terminal token IDs are recognized correctly. Cases such as `[_CHUNK_EOS_ID]`, `[turn_eos_token_id]`, and a listen token followed by an end marker would strengthen coverage of the control/terminal scenarios described in the PR. These should still return `[]` when neither latent nor hidden states is present.

File: tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+def test_native_duplex_speak_without_hidden_states_still_raises() -> None:
+    """A speak unit still fails closed when speech conditioning is missing.
+
+    Skip applies only to confirmed control-only/no-speech handoffs. If the
+    Thinker emitted speak/content tokens, missing latent/hidden_states is a
+    malformed speech handoff and must keep raising.
+    """
+    with pytest.raises(ValueError, match="No latent or hidden_states"):
```
Comment (amy-why-3459) [inline tests/model_executor/models/minicpmo_4_5/test_llm2tts_native_duplex_silence.py:158]: Non-blocking: please consider a contrast case that verifies the skip remains restricted to native duplex.

The speak case protects against dropping malformed speech handoffs, but all four new cases carry native-duplex metadata. An ordinary request with no native-duplex marker, empty output token IDs, and no latent/hidden states should still raise `ValueError`. That case would catch accidentally removing the `is_native_duplex_handoff` condition and silently swallowing malformed ordinary requests. The existing adjacent missing-hidden-state test uses a non-empty content token, so it does not cover this particular boundary.

**Review by amy-why-3459 — APPROVED** [review]
The four tests match the existing silence-skip behavior, and I found no blocking issues. The cumulative speak-then-listen case is particularly useful: it guards against checking the entire cumulative token sequence instead of the current unprocessed segment.

Two non-blocking coverage suggestions are inline.

Validation at 9c1a74ae44278627ac5a0e8b7c9f83155503e776: the new file passed all 4 tests. Running it together with the adjacent test_llm2tts.py suite produced 23 passed and 1 skipped. I did not independently rerun the author's pre-fix-commit verification.

**Comment by ZacheryAU** [comment]
> The four tests match the existing silence-skip behavior, and I found no blocking issues. The cumulative speak-then-listen case is particularly useful: it guards against checking the entire cumulative token sequence instead of the current unprocessed segment.
> 
> Two non-blocking coverage suggestions are inline.
> 
> Validation at [9c1a74a](https://github.com/vllm-project/vllm-omni/commit/9c1a74ae44278627ac5a0e8b7c9f83155503e776): the new file passed all 4 tests. Running it together with the adjacent test_llm2tts.py suite produced 23 passed and 1 skipped. I did not independently rerun the author's pre-fix-commit verification.

Thanks for the suggestions and they are both added
